# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.h
# Commit: 2fd61eb5c69c
# Full Hash: 2fd61eb5c69ce9ac806048a35c7a7a88bf4b9652
# Author: Perry Jiang <perry@mozilla.com>
# Date: 2020-04-21 09:42:20
# Regressor Bug: 1618546
# File Overlap Count: 2
# Description:
#   Bug 1618546 - give worker debugger globals their own clients r=asuth,webidl,smaug
#   
#   - Worker debugger globals gets a client with a null principal
#   - Ensure globals are created before script loads
#   - Introduce WorkerGlobalScopeBase to share code
# ==============================================================================

diff -r 9ccf9f680b64 -r 2fd61eb5c69c dom/workers/ScriptLoader.h
--- a/dom/workers/ScriptLoader.h	Tue Apr 21 06:09:28 2020 +0000
+++ b/dom/workers/ScriptLoader.h	Tue Apr 21 06:50:53 2020 +0000
@@ -8,14 +8,15 @@
 #define mozilla_dom_workers_scriptloader_h__
 
 #include "mozilla/dom/WorkerCommon.h"
+#include "nsIContentPolicy.h"
 #include "nsStringFwd.h"
 
-class nsIPrincipal;
-class nsIURI;
-
-class nsILoadGroup;
 class nsIChannel;
 class nsICookieJarSettings;
+class nsILoadGroup;
+class nsIPrincipal;
+class nsIReferrerInfo;
+class nsIURI;
 
 namespace mozilla {
 
@@ -23,6 +24,8 @@
 
 namespace dom {
 
+class ClientInfo;
+class Document;
 struct WorkerLoadInfo;
 class WorkerPrivate;
 class SerializedStackHolder;